app/lib/shared/handler_helpers.dart

```diff
@@ -49,6 +50,24 @@ Future<void> runHandler(
   }, shared: true);
 }
 
+// Extends response with content security policy headers when the response's
+// content type is HTML.
+shelf.Handler _cspHeaderWrapper(shelf.Handler handler) {
+  return (shelf.Request request) async {
+    final rs = await handler(request);
+    final contentType = rs.headers['content-type'];
+    final isHtml = contentType != null && contentType.startsWith('text/html');
+    if (isHtml) {
+      return rs.change(headers: {
+        'x-content-type-options': 'nosniff',
+        'content-security-policy': contentSecurityPolicy,
+      });
```

Comment:
Don't we also need something about frame options to protect against click jacking in old browser..
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options

Reply:
`X-Frame-Options: sameorigin`?

Reply:
do we use frames anywhere? If not we can just forbid it, and allow it when we need to...

Reply:
ack, `deny` it is.